### ISSUE TITLE
Inconsistent naming between `Data.(Nat/Integer).Properties`

### ISSUE THREAD

**Issue by MatthewDaggitt** [agda/agda-stdlib#1579]
We have our left and rights switched round in `*-monoʳ-≤`, `*-monoʳ-≤-pos`. It would be kind of good if we had definitions for `Monotonic`/`LeftMonotonic`/`RightMonotonic` etc. in `Relation.Binary.Definitions` to help keep us on the straight and narrow.

**Comment by MatthewDaggitt** [comment]
Also `Antimonotonic` etc.

**Comment by jamesmckinna** [comment]
See also #2087 / #2089

**Comment by jamesmckinna** [comment]
Regarding `Anti`, I have the (slight? vague?) feeling that as with `Cancellative`, it might be better to parametrise these properties on 'source' and 'target' relations... cf. `Preserves`?

**Comment by jamesmckinna** [comment]
> Also `Antimonotonic` etc.

Are we sure we need this? Monotonicity is/should be usually stated wrt two *distinct* orderings, and these can then be varied as:
* same, same: 'usual' monotonicity within a given ordered structure
* same, flipped: antitone
* etc.

**Comment by jamesmckinna** [comment]
See also #2532 for the wider issue of the (need to agree the) semantics of 'left' and 'right'...

**Comment by jamesmckinna** [comment]
Aaaargh! The various definitions *do* exist in `Relation.Binary.Definitions`, but aren't then specialised in `Algebra.Definitions` (and perhaps that wouldn't make sense as such to do so...?).

**Comment by JacquesCarette** [comment]
My guess why it's this way: history. Things were developed "by hand" and only slowly did patterns emerge. You could say that those patterns should be well-known - and indeed they are to the people whose educational path included enough of the right kind of Algebra. But that doesn't tend to be the same people as those who develop ITPs! [I've seen the same thing happen in CASes, but there it's mostly about not knowing enough analysis.]

**Comment by jamesmckinna** [comment]
> [#2580](https://github.com/agda/agda-stdlib/pull/2580) shouldn't close this issue right?

No, you're correct, but it might do once that DRAFT PR is finished? 
In any case, will tweak the preamble to it to remove the `fixes` cue for the time being...

**Comment by jamesmckinna** [comment]
Progress so far on #2580 suggests that at least the various `Data.X.Properties` module are *mutually* consistent, in naming lemmas vs. the 'opposite' handedness in the lemma statements... so we should only consider a `breaking` change *after* resolving the discussion on #2532 ?
